Startup/startup_ch32v20x_D6.S

```diff
@@ -151,141 +138,117 @@ _vector_base:
-TIM4_IRQHandler:  1:  j 1b
-I2C1_EV_IRQHandler:  1:  j 1b
-I2C1_ER_IRQHandler:  1:  j 1b
-I2C2_EV_IRQHandler:  1:  j 1b
-I2C2_ER_IRQHandler:  1:  j 1b
-SPI1_IRQHandler:  1:  j 1b
-SPI2_IRQHandler:  1:  j 1b
-USART1_IRQHandler:  1:  j 1b
-USART2_IRQHandler:  1:  j 1b
-USART3_IRQHandler:  1:  j 1b
-EXTI15_10_IRQHandler:  1:  j 1b
-RTCAlarm_IRQHandler:  1:  j 1b
-USBWakeUp_IRQHandler:  1:  j 1b
-USBHD_IRQHandler:  1:  j 1b
-USBHDWakeUp_IRQHandler:  1:  j 1b
-UART4_IRQHandler:  1:  j 1b
-DMA1_Channel8_IRQHandler:  1:  j 1b
+NMI_Handler:
+HardFault_Handler:
+Ecall_M_Mode_Handler:
+Ecall_U_Mode_Handler:
+Break_Point_Handler:
+SysTick_Handler:
+SW_Handler:
+WWDG_IRQHandler:
+PVD_IRQHandler:
+TAMPER_IRQHandler:
+RTC_IRQHandler:
+FLASH_IRQHandler:
+RCC_IRQHandler:
+EXTI0_IRQHandler:
+EXTI1_IRQHandler:
+EXTI2_IRQHandler:
+EXTI3_IRQHandler:
+EXTI4_IRQHandler:
+DMA1_Channel1_IRQHandler:
+DMA1_Channel2_IRQHandler:
+DMA1_Channel3_IRQHandler:
+DMA1_Channel4_IRQHandler:
+DMA1_Channel5_IRQHandler:
+DMA1_Channel6_IRQHandler:
+DMA1_Channel7_IRQHandler:
+ADC1_2_IRQHandler:
+USB_HP_CAN1_TX_IRQHandler:
+USB_LP_CAN1_RX0_IRQHandler:
+CAN1_RX1_IRQHandler:
+CAN1_SCE_IRQHandler:
+EXTI9_5_IRQHandler:
+TIM1_BRK_IRQHandler:
+TIM1_UP_IRQHandler:
+TIM1_TRG_COM_IRQHandler:
+TIM1_CC_IRQHandler:
+TIM2_IRQHandler:
+TIM3_IRQHandler:
+TIM4_IRQHandler:
+I2C1_EV_IRQHandler:
+I2C1_ER_IRQHandler:
+I2C2_EV_IRQHandler:
+I2C2_ER_IRQHandler:
+SPI1_IRQHandler:
+SPI2_IRQHandler:
+USART1_IRQHandler:
+USART2_IRQHandler:
+USART3_IRQHandler:
+EXTI15_10_IRQHandler:
+RTCAlarm_IRQHandler:
+USBWakeUp_IRQHandler:
+USBFS_IRQHandler:
+USBFSWakeUp_IRQHandler:
+UART4_IRQHandler:
+DMA1_Channel8_IRQHandler:
+1:
+	j 1b
 
-    .section	.text.handle_reset,"ax",@progbits
-    .weak	handle_reset
-    .align	1
+	.section	.text.handle_reset,"ax",@progbits
+	.weak	handle_reset
+	.align	1
 handle_reset:
 .option push 
 .option	norelax 
-    la gp, __global_pointer$
+	la gp, __global_pointer$
 .option	pop 
 1:
-    la sp, _eusrstack 
+	la sp, _eusrstack 
 2:
-    /* Load data section from flash to RAM */
-    la a0, _data_lma
-    la a1, _data_vma
-    la a2, _edata
-    bgeu a1, a2, 2f
+/* Load data section from flash to RAM */
+	la a0, _data_lma
+	la a1, _data_vma
+	la a2, _edata
+	bgeu a1, a2, 2f
 1:
-    lw t0, (a0)
-    sw t0, (a1)
-    addi a0, a0, 4
-    addi a1, a1, 4
-    bltu a1, a2, 1b
+	lw t0, (a0)
+	sw t0, (a1)
+	addi a0, a0, 4
+	addi a1, a1, 4
+	bltu a1, a2, 1b
 2:
-    /* Clear bss section */
-    la a0, _sbss
-    la a1, _ebss
-    bgeu a0, a1, 2f
+/* Clear bss section */
+	la a0, _sbss
+	la a1, _ebss
+	bgeu a0, a1, 2f
 1:
-    sw zero, (a0)
-    addi a0, a0, 4
-    bltu a0, a1, 1b
+	sw zero, (a0)
+	addi a0, a0, 4
+	bltu a0, a1, 1b
 2:
+/* Configure pipelining and instruction prediction */
     li t0, 0x1f
     csrw 0xbc0, t0
-
-    #if defined(__PIO_BUILD_FREERTOS__) || defined(__PIO_BUILD_HARMONY_LITEOS__) || defined(__PIO_BUILD_RT_THREAD__) || defined(__PIO_BUILD_TENCENT_OS__)
```

Comment:
Killing these changes will cause build failures for FreeRTOS, LiteOS, RT-Thread and Tencent-OS, I really need those to be left in.

Reply:
that make sense, I will change, just want to highlight that:

```diff
- csrs mstatus, t0
+ csrw mstatus, t0
```

I'm not really familiar with this instructions, but does it matter?

Reply:
Wow that's really a bug in the old version. They definitely intended to do a *write* to the constrol status register, not *read* from it.

This assembly file looks good now. But, next to this startup_ch32v20x_D6.S file, the startup_ch32v20x_D8.S and startup_ch32v20x_D8W.S file would need the same updating?

Reply:
I didn’t touch those files but I understand that the bug may be present there also, I’ll check soon

Reply:
I have fixed for the other CH32V2 variants, however it seems like the problem is also on other variants, for example https://github.com/Community-PIO-CH32V/framework-wch-noneos-sdk/blob/5992d6957398992f7455e614c4988458f1aa970a/Startup/startup_ch32v30x_D8C.S#L368

I think if those are to be fixed, it should be done on another PR for clarity

Reply:
Mhm I thought we could copy the patches from
* https://github.com/openwch/ch32v20x/blob/main/EVT/EXAM/SRC/Startup/startup_ch32v20x_D8.S
* https://github.com/openwch/ch32v20x/blob/main/EVT/EXAM/SRC/Startup/startup_ch32v20x_D8W.S

Reply:
What do you mean? Those files are already copied over.

Reply:
No when I compare 

https://github.com/openwch/ch32v20x/blob/main/EVT/EXAM/SRC/Startup/startup_ch32v20x_D8.S#L158-L218 

and the state of the PR

https://github.com/ldab/framework-wch-noneos-sdk/blob/ch32v20x/Startup/startup_ch32v20x_D8.S#L171-L229

they're different still-

Reply:
my bad, I can see it now. added!